File: src/models/forwarding.ts
export type ForwardKind = "local" | "remote" | "dynamic";

export interface PortForward {
  id: string;
  sessionId: string;
  kind: ForwardKind;
  localHost: string;
  localPort: number;
  remoteHost: string;
  remotePort: number;
  description?: string | null;
}

export interface StartPortForwardRequest {
  sessionId: string;
  profileId: string;
  kind: ForwardKind;
  localHost: string;
  localPort: number;
  remoteHost: string;
  remotePort: number;
  description?: string | null;
}

File: src-tauri/src/models/forwarding.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, PartialEq, Eq, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub enum ForwardKind {
    Local,
    Remote,
    Dynamic,
}

#[derive(Debug, Clone, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct PortForward {
    pub id: String,
    pub session_id: String,
    pub kind: ForwardKind,
    pub local_host: String,
    pub local_port: u16,
    pub remote_host: String,
    pub remote_port: u16,
    pub description: Option<String>,
}

#[derive(Debug, Clone, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct StartPortForwardRequest {
    pub session_id: String,
    pub profile_id: String,
    pub kind: ForwardKind,
    pub local_host: String,
    pub local_port: u16,
    pub remote_host: String,
    pub remote_port: u16,
    pub description: Option<String>,
}
